rclpy/rclpy/task.py

```diff
@@ -128,13 +126,14 @@ def set_exception(self, exception):
             self._cancelled = False
         self._schedule_or_invoke_done_callbacks()
 
-    def _schedule_or_invoke_done_callbacks(self):
+    def _schedule_or_invoke_done_callbacks(self) -> None:
         """
         Schedule done callbacks on the executor if possible, else run them directly.
 
         This function assumes self._lock is not held.
         """
         with self._lock:
+            assert self._executor is not None
```

Comment:
this check is required here? i think this could be `None`.

Reply:
I believe if `self._executor` is `None` then an error of  `'NoneType' object is not callable` would occur with the line `executor = self._executor()`